### PR TITLE
Add names to child DVGeos

### PR DESCRIPTION
**Pull request by anilyil**
## Purpose

This PR changes how DVGeo class tracks its children. Old implementation just appended the children to a list, which is then indexed in order. The current implementation saves the child DVGeometries in the `self.children` dictionary. The keys of the dictionary are the names of each child DVGeo. This makes the code a lot more readable since each child DVGeo is referenced by its name. 

The old functionality is maintained, and unless users were accessing the child DVGeometries in a hacky way by indexing the old list, everything should work fine. Some output names might be different, but we can also fix this behavior. The names of the added pointsets will be the same if users don't provide a name for the child DVGeo.

I will apply a similar set of changes to the mphys wrapper once #193 is merged. 

## Expected time until merged

2 weeks

## Type of change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [ ] Maintenance update
- [ ] Other (please describe)

## Checklist

- [ ] I have run `flake8` and `black` to make sure the Python code adheres to PEP-8 and is consistently formatted
- [ ] I have formatted the Fortran code with `fprettify` or C/C++ code with `clang-format` as applicable
- [ ] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

File: pygeo/parameterization/DVGeo.py [hunk 1]
```diff
@@ -752,6 +752,11 @@ def coordXfer(coords, mode="fwd", applyDisplacement=True, **kwargs):
         # compNames is only needed for DVGeometryMulti, so remove it if passed
         kwargs.pop("compNames", None)
 
+        # check if we want a custom subset of child DVGeos
```
Comment (marcomangano) [inline pygeo/parameterization/DVGeo.py:755]: This looks like an interesting addition, could you please add a short description of the arg and intended functionality in the docstrings for `addPointSet`?

**Review by marcomangano — COMMENTED** [review]
I gave a quick pass and it looks ok to me, see my comment below. Maybe we should also add a test for `activeChildren`?

**Review by bernardopacini — APPROVED** [review]
The changes here look good to me, just a question regarding compatibility:

- How does this affect other DVGeo variants such as DVGeoMulti or DVGeoAxi? If I am not mistaken, it should be fine because they inherit this base anyways, right?

**Comment by hajdik** [comment]
> Should we change the approach here to rely on that name attribute?

Yes, there is a `name` attribute in the BaseDVGeo class so everything inheriting from that has access to it, I think it makes sense to use that since it's already there.

File: tests/reg_tests/test_DVGeometry.py [hunk 2]
```diff
@@ -1460,6 +1460,72 @@ def test_shape_functions(self, train=False, refDeriv=False):
...
+                dIdPt = np.zeros([nPt, 2, 3])
+                dIdPt[0, 0, 0] = 1.0
+                dIdPt[1, 0, 1] = 1.0
+                dIdPt[2, 0, 2] = 1.0
+                dIdPt[3, 1, 0] = 1.0
+                dIdPt[4, 1, 1] = 1.0
+                dIdPt[5, 1, 2] = 1.0
+                dIdx = DVGeo.totalSensitivity(dIdPt, ptName)
+
+                handler.root_add_dict(f"dIdx_{ptName}", dIdx, rtol=1e-10, atol=1e-10)
```
Comment (hajdik) [inline tests/reg_tests/test_DVGeometry.py:1517]: Looks like the test is failing here with `KeyError: 'child1_span1'` but once that's fixed I'm good with this PR

File: pygeo/parameterization/DVGeo.py [hunk 3]
```diff
@@ -908,17 +941,28 @@ def addGlobalDV(self, dvName, value, func, lower=None, upper=None, scale=1.0, co
             Use a string for a single configuration or a list for multiple
             configurations. The default value of None implies that the design
             variable applies to *ALL* configurations.
+        prependName : bool
+            Flag to determine if self.name attribute is prepended to this DV name.
```
Comment (hajdik) [inline pygeo/parameterization/DVGeo.py:945]: Can you add something in the docstring about why this option exists?

**Review by hajdik — APPROVED** [review]

**Pull request merged**